src/header/Header.js

```diff
@@ -1,46 +1,70 @@
 import React from 'react';
 import { Link } from 'react-router-dom';
-import { compose } from 'react-apollo';
-import Login from './Login';
 import AppBar from '@material-ui/core/AppBar';
 import Toolbar from '@material-ui/core/Toolbar';
+import Button from '@material-ui/core/Button';
 import Grid from '@material-ui/core/Grid';
 import logo from '../images/cfd-circle-icon-white.png';
 import userIsAdmin from '../utils/userIsAdmin';
-import withLoggedInUser from '../utils/withLoggedInUser';
-import withAuthSession from '../utils/withAuthSession';
+import MenuList from './Menu';
+import AuthenticationContext from '../utils/authentication/authContext';
+import AuthService from '../utils/authentication/authService';
 import './Header.css';
 
-const Header = ({ isAuthenticated, user }) => {
+const Header = () => {
   return (
-    <AppBar position="fixed">
-      <Toolbar>
-        <Link to="/">
-          <img className="Header-logo" src={logo} alt="code for denver logo" />
-        </Link>
-        <Grid container justify="space-between" alignItems="center">
-          {isAuthenticated && (
-            <Grid item>
-              <Link className="Header-link" to="/volunteers">
-                All Users
-              </Link>
-              <Link className="Header-link" to="/projects">
-                All Projects
-              </Link>
-              {userIsAdmin(user) && (
-                <Link className="Header-link" to="/admin">
-                  Admin Resources
-                </Link>
+    <AuthenticationContext.Consumer>
+      {context => (
+        <AppBar position="fixed">
+          <Toolbar>
+            <Link to="/">
+              <img
+                className="Header-logo"
+                src={logo}
+                alt="code for denver logo"
+              />
+            </Link>
+            <Grid container justify="space-between" alignItems="center">
+              {context.isAuthenticated() ? (
+                <React.Fragment>
+                  <Grid item>
+                    <Link className="Header-link" to="/volunteers">
+                      All Users
+                    </Link>
+                    <Link className="Header-link" to="/projects">
+                      All Projects
+                    </Link>
+                    {userIsAdmin(context.authData.role) && (
+                      <Link className="Header-link" to="/admin">
+                        Admin Resources
+                      </Link>
+                    )}
+                  </Grid>
+                  <MenuList
+                    avatar={context.authData.userProfile.picture}
+                    username={context.authData.userProfile.name}
+                    logout={() => AuthService.logout()}
+                  />
+                </React.Fragment>
+              ) : (
+                <Grid item>
+                  <Button color="secondary" onClick={() => AuthService.login()}>
```

Comment:
No need for the arrow function, just use `onClick={AuthService.login}`. Man this is so much nicer!